onnxruntime/core/framework/mem_pattern_planner.h

```diff
@@ -31,7 +31,38 @@ class MemPatternPlanner {
  public:
   MemPatternPlanner() = default;
 
-  void TraceAllocation(int ml_value_idx, size_t size) {
+  // Returns true if there is an intersection between two time schedules.
+  // ASSUMES EACH TIME SCHEDULE IS SORTED. THIS IS VALIDATED AT THE END OF MEMORY PLANNING.
+  bool OverlappingTimeSchedules(const std::vector<size_t>& program_counter_start_1, const std::vector<size_t>& program_counter_end_1,
```

Comment:
(nit):  you could make these a `std::vector<std::pair<size_t, size_t>>` instead of 2 parallel vectors for start and end.